test/hexpm/web/views/package_view_test.exs

```diff
@@ -107,4 +107,51 @@ defmodule Hexpm.Web.PackageViewTest do
     assert PackageView.snippet_version(:erlang_mk, version3) == "2.0.2"
   end
 
+  describe "retirement_message/1" do
+    test "reason is 'other', message contains text" do
+      retirement = %{reason: "other", message: "something went terribly wrong"}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "something went terribly wrong"]
+    end
+
+    test "reason is 'other', message is empty" do
+      retirement = %{reason: "other", message: nil}
+      assert PackageView.retirement_message(retirement) == ["Retired package"]
+    end
+
+    test "reason is not 'other', message contains text" do
+      retirement = %{reason: "security", message: "something went terribly wrong"}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "Security issue", " - ", "something went terribly wrong"]
+    end
+
+    test "reason is not 'other', message is empty" do
+      retirement = %{reason: "security", message: nil}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "Security issue"]
+    end
+  end
+
+  describe "retirement_html/1" do
+    test "reason is 'other', message contains text" do
+      retirement = %{reason: "other", message: "something went terribly wrong"}
+      assert PackageView.retirement_html(retirement) ==
+        [{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]}, " reason: ", "something went terribly wrong"]
```

Comment:
You can use `Phoenix.HTML.safe_to_string/1` to return a proper string. This way we don't rely on phoenix internals.

Reply:
Sorry, I am not as familiar with PHX. From what I can tell in the documentation, I would apply that function to `{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]}`, which means it should be in the `retirement_html/1` function, not in the test. Is that correct?

Reply:
or, i could map the `PackageView.retirement_html(retirement)` result and apply the function to each element on the array, and then join it here. TBH - I am not sure what the template is expecting.

Reply:
The template expects `{:safe, ...}`. You should call `Phoenix.HTML.safe_to_string/1` in the test.